jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -514,129 +518,174 @@ def _get_source_identifiers_by_hdu(hdus, filepaths=None, header_keys=['SOURCEID'
+            except TypeError:
+                dispersion = None
+
+            # distinguish image files from counts files -- only the former go
+            # through the "Assign World Coordinate System" calibration step
+            s_wcs = header.get('S_WCS')
+
+            if datamodl is None:
+                continue
+            if datamodl == 'MultiSpecModel' and dispersion == 'C':
+                file_lists['1D Spectra C'].append(filepath)
+            elif datamodl == 'MultiSpecModel' and dispersion == 'R':
+                file_lists['1D Spectra R'].append(filepath)
+            elif datamodl == 'MultiSlitModel' and dispersion == 'C':
+                file_lists['2D Spectra C'].append(filepath)
+            elif datamodl == 'MultiSlitModel' and dispersion == 'R':
+                file_lists['2D Spectra R'].append(filepath)
+            elif datamodl == 'ImageModel' and s_wcs is not None:
+                file_lists['Direct Image'].append(filepath)
+            else:
+                continue
 
-    # Retrieve source information
-    for source_catalog_num in range(0, len(file_lists["Source Catalog"])):
-        cat_file = file_lists["Source Catalog"][source_catalog_num]
-        parsed_cat_fields = _fields_from_ecsv(cat_file, cat_fields, delimiter=" ")
-        pupil = [x
-                 for x in cat_file.split("/")[-1].split("_")
-                 if x[0] == "F" or x[0] == "f"][0]
+        elif filepath.suffix == '.ecsv':
+            if file_lists[cat_key]:
+                raise ValueError("source directory has more "
+                                 "than one source catalog")
+            else:
+                file_lists[cat_key] = filepath
 
-        pupil_id_dict[pupil] = {}
+        else:
+            continue
 
-        for row in parsed_cat_fields:
-            pupil_id_dict[pupil][int(row[0])] = (row[1], row[2])
+    # validate that all expected files are present in proper amounts
+    _warn_if_not_found(app, file_lists)
 
-    # Read in direct image filters
-    image_dict = {}
-    filter_wcs = {}
+    # Parse relevant information from source catalog
+    cat_id_dict = {}
+    cat_file = file_lists[cat_key]
+    try:
+        cat_fields = ['label', 'sky_centroid.ra', 'sky_centroid.dec']
+        parsed_cat_fields = _fields_from_ecsv(cat_file, cat_fields,
+                                              delimiter=" ")
+    except KeyError:
+        # Older pipeline builds use different colname to distinguish sources
+        cat_fields[0] = 'id'
+        parsed_cat_fields = _fields_from_ecsv(cat_file, cat_fields,
+                                              delimiter=" ")
+    for row in parsed_cat_fields:
+        cat_id_dict[int(row[0])] = (row[1], row[2])
 
     # Set up a dictionary of datasets to add to glue
     add_to_glue = {}
 
-    print("Loading: Images")
+    # Parse images
+    image_dict = {}
+    filter_wcs = {} # NEEDED? HAS VALUE ASSIGNED LATER BUT NEVER USED OTHERWISE
```

Comment:
I found where [this](https://github.com/spacetelescope/jdaviz/commit/505edfe0128a61d46cee1599c436a76d1b011285#diff-9dac277cb692b00ef5b97d4aedcc083497fda675acdec85547c34a7ec11a3c1bL440) used to be used. This is back when we would add data from the SCI extension of the 2d spectra, but for some reason the WCS information was in another file (image maybe? Can't remember). I think this can safely be removed now since we load data another way.